File: iOS/Spaces V1.5/Services/DeviceIdentityService.swift
import Foundation

struct DeviceEncryptionIdentity: Hashable {
    let deviceID: String
    let platform: String
    let publicKey: String
}

final class DeviceIdentityService {
    private let defaults: UserDefaults
    private let key = "spaces.device.id"

    init(defaults: UserDefaults = .standard) {
        self.defaults = defaults
    }

    func currentDeviceID() -> String {
        if let existing = defaults.string(forKey: key)?.trimmingCharacters(in: .whitespacesAndNewlines),
           !existing.isEmpty {
            return existing
        }

        let newID = "ios-" + UUID().uuidString.lowercased()
        defaults.set(newID, forKey: key)
        return newID
    }

    func currentPlatform() -> String {
        "ios"
    }
}

File: Android V1.6/app/src/main/java/com/arcinteractive/spaces/data/auth/DeviceIdentityService.kt
package com.arcinteractive.spaces.data.auth

import android.content.Context
import java.util.UUID

data class DeviceEncryptionIdentity(
    val deviceId: String,
    val platform: String,
    val publicKey: String
)

class DeviceIdentityService {
    fun currentDeviceId(context: Context): String {
        val preferences = context.getSharedPreferences("spaces_device_identity", Context.MODE_PRIVATE)
        val existing = preferences.getString("device_id", null)?.trim().orEmpty()
        if (existing.isNotEmpty()) {
            return existing
        }

        val newId = "android-${UUID.randomUUID().toString().lowercase()}"
        preferences.edit().putString("device_id", newId).apply()
        return newId
    }

    fun currentPlatform(): String {
        return "android"
    }
}
